### PR TITLE
MAGECLOUD-3816: Eliminate Docker config.env file

### PR DESCRIPTION
**Pull request by shiftedreality**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-3816

### Manual testing scenarios

1. https://jira.corp.magento.com/browse/MAGECLOUD-66
2. https://jira.corp.magento.com/browse/MAGECLOUD-116

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Command/Docker/ConfigConvert.php [hunk 1]
```diff
@@ -1,108 +0,0 @@
...
-use Symfony\Component\Console\Command\Command;
-use Symfony\Component\Console\Input\InputInterface;
-use Symfony\Component\Console\Output\OutputInterface;
-
-/**
- * Converts raw array configuration to .env files.
- */
-class ConfigConvert extends Command
-{
-    const NAME = 'docker:config:convert';
```
Comment (NadiyaS) [inline src/Command/Docker/ConfigConvert.php:22]: you remove this command at all?
if so, it is BIC. You need to leave this command and result of it should be output with explanation what to use instead

File: src/Command/Docker/GenerateDist.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @inheritdoc
+     */
+    protected function configure()
+    {
+        $this->setName(self::NAME)
+            ->setDescription('Generates Docker .dist files');
+    }
+
+    /**
+     * @inheritDoc
```
Comment (NadiyaS) [inline src/Command/Docker/GenerateDist.php]: cover @inheritdoc with {}

File: src/Docker/Config/Environment/Reader.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace Magento\MagentoCloud\Docker\Config\Environment;
+
+use Magento\MagentoCloud\Docker\ConfigurationMismatchException;
+use Magento\MagentoCloud\Filesystem\DirectoryList;
+use Magento\MagentoCloud\Filesystem\Driver\File;
+use Magento\MagentoCloud\Filesystem\SystemList;
+
+/**
+ * Reader for .env files.
```
Comment (NadiyaS) [inline src/Docker/Config/Environment/Reader.php]: Seems like description is not correct

File: src/Command/Docker/GenerateDist.php (hunk 2 above)
Comment (shiftedreality) [reply src/Command/Docker/GenerateDist.php]: Fixed

**Review by NadiyaS — CHANGES_REQUESTED** [review]
We need to define the BC rules.
Cause what I see now is:
 - removing CLI command
 - removing exception type

what is BIC as for me, especially the first one

File: src/Command/Docker/ConfigConvert.php (hunk 1 above)
Comment (shiftedreality) [reply src/Command/Docker/ConfigConvert.php:22]: It is a manual run command, which can be excluded from BC list

**Comment by shiftedreality** [comment]
Exceptions in ECE-Tools are not part of BC rules

File: src/Command/Docker/GenerateDist.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+     */
+    public function __construct(Generator $distGenerator)
+    {
+        $this->distGenerator = $distGenerator;
+
+        parent::__construct();
+    }
+
+    /**
+     * {@inheritDoc}
```
Comment (BaDos) [inline src/Command/Docker/GenerateDist.php]: You should use `{}` only if you expand the description.

For example:
```
/**
 * @inheritDoc
 */
```

```
/**
 * {@inheritDoc}
 * @return bool
 */
```

Comment (BaDos) [reply src/Command/Docker/GenerateDist.php]: Please, fix all places with `@inheritDoc`

Comment (BaDos) [reply src/Command/Docker/GenerateDist.php]: Also, I see that usually, we use `@inheritdoc`, you write `@inheritDoc`
I think we should use one style.

**Review by BaDos — CHANGES_REQUESTED** [review]
Please, fix all @inheritdoc

File: src/Command/Docker/GenerateDist.php (hunk 4 above)
Comment (shiftedreality) [reply src/Command/Docker/GenerateDist.php]: There is no clear explanation of `@inheritDoc`, until that, I don't see a big reason to point attention on it.

File: src/Command/Docker/ConfigConvert.php (hunk 1 above)
Comment (NadiyaS) [reply src/Command/Docker/ConfigConvert.php:22]: hm. Manual command may be used in scripts by others to automate deployment on cloud docker, for example. And even if use it only manually, it is not good strategy to surprise people with absence of some commands after new release.
I agree with removing Extension in current moment as we do not provide BC for code itself, but removing/renaming commands or change their behavior is BIC IMHO.

Comment (shiftedreality) [reply src/Command/Docker/ConfigConvert.php:22]: Added name alias

**Review by BaDos — DISMISSED** [review]

**Review by NadiyaS — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Comment by andriyShevtsov** [comment]
QA approved

**Pull request merged**